app/helpers/elasticsearch_helper.rb

```diff
@@ -32,11 +33,31 @@ def taxon_search(prefix, tax_levels = TaxonCount::NAME_2_LEVEL.keys)
         }
       end
     end
-    matching_taxa.values
+
+    if project_id.present?
+      filter_by_project(matching_taxa.values, project_id)
+    else
+      matching_taxa.values
+    end
```

Comment:
Do you know how many results elastic search returns?
What would happen if none of the results were in the project?

Reply:
I guess it would just have to wait for the next character for more suggestions?